### ISSUE TITLE
`noSecrets` False positive secret detection

### ISSUE THREAD

**Issue by lloydjatkinson** [biomejs/biome#8809]
### Environment information

sprinkles.css.ts:

```bash

shorthands: {
    padding: ['paddingTop', 'paddingBottom', 'paddingLeft', 'paddingRight'],
    paddingX: ['paddingLeft', 'paddingRight'],
    paddingY: ['paddingTop', 'paddingBottom'],
},


// Color properties
const colorProperties = defineProperties({
    properties: {
        color,
        backgroundColor: color,
        borderColor: color,
    },
    shorthands: {
        bg: ['backgroundColor'],
    },
});
```

`paddingBottom`, `paddingRight`, `backgroundColor` are detected as a potential secret. Typical usage of https://vanilla-extract.style/.

Can't reproduce it in playground but here is a screenshot.

<img width="781" height="876" alt="Image" src="https://github.com/user-attachments/assets/381c8cef-3987-488e-9a60-70cde41e0b9e" />

### Rule name

lint/security/noSecrets

### Playground link

https://biomejs.dev/playground/?quoteStyle=single&indentWidth=4&lintRules=all&code=YwBvAG4AcwB0ACAAcgBlAHMAcABvAG4AcwBpAHYAZQBQAHIAbwBwAGUAcgB0AGkAZQBzACAAPQAgAGQAZQBmAGkAbgBlAFAAcgBvAHAAZQByAHQAaQBlAHMAKAB7AAoAIAAgACAAIABkAGUAZgBhAHUAbAB0AEMAbwBuAGQAaQB0AGkAbwBuADoAIAAnAG0AbwBiAGkAbABlACcALAAKACAAIAAgACAAYwBvAG4AZABpAHQAaQBvAG4AcwA6ACAAewAKACAAIAAgACAAIAAgACAAIABtAG8AYgBpAGwAZQA6ACAAewB9ACwACgAgACAAIAAgACAAIAAgACAAdABhAGIAbABlAHQAOgAgAHsAIAAnAEAAbQBlAGQAaQBhACcAOgAgAG0AZQBkAGkAYQAuAHQAYQBiAGwAZQB0ACAAfQAsAAoAIAAgACAAIAAgACAAIAAgAGQAZQBzAGsAdABvAHAAOgAgAHsAIAAnAEAAbQBlAGQAaQBhACcAOgAgAG0AZQBkAGkAYQAuAGQAZQBzAGsAdABvAHAAIAB9ACwACgAgACAAIAAgAH0ALAAKACAAIAAgACAAcgBlAHMAcABvAG4AcwBpAHYAZQBBAHIAcgBhAHkAOgAgAFsAJwBtAG8AYgBpAGwAZQAnACwAIAAnAHQAYQBiAGwAZQB0ACcALAAgACcAZABlAHMAawB0AG8AcAAnAF0ALAAKACAAIAAgACAAcAByAG8AcABlAHIAdABpAGUAcwA6ACAAewAKACAAIAAgACAAIAAgACAAIABkAGkAcwBwAGwAYQB5ADoAIABbACcAbgBvAG4AZQAnACwAIAAnAGIAbABvAGMAawAnACwAIAAnAGYAbABlAHgAJwAsACAAJwBnAHIAaQBkACcALAAgACcAaQBuAGwAaQBuAGUAJwAsACAAJwBpAG4AbABpAG4AZQAtAGYAbABlAHgAJwBdACwACgAgACAAIAAgACAAIAAgACAAZgBsAGUAeABEAGkAcgBlAGMAdABpAG8AbgA6ACAAWwAnAHIAbwB3ACcALAAgACcAYwBvAGwAdQBtAG4AJwBdACwACgAgACAAIAAgACAAIAAgACAAYQBsAGkAZwBuAEkAdABlAG0AcwA6ACAAWwAnAHMAdAByAGUAdABjAGgAJwAsACAAJwBmAGwAZQB4AC0AcwB0AGEAcgB0ACcALAAgACcAYwBlAG4AdABlAHIAJwAsACAAJwBmAGwAZQB4AC0AZQBuAGQAJwBdACwACgAgACAAIAAgACAAIAAgACAAagB1AHMAdABpAGYAeQBDAG8AbgB0AGUAbgB0ADoAIABbACcAZgBsAGUAeAAtAHMAdABhAHIAdAAnACwAIAAnAGMAZQBuAHQAZQByACcALAAgACcAZgBsAGUAeAAtAGUAbgBkACcALAAgACcAcwBwAGEAYwBlAC0AYgBlAHQAdwBlAGUAbgAnAF0ALAAKACAAIAAgACAAIAAgACAAIABnAGEAcAA6ACAAcwBwAGEAYwBlACwACgAgACAAIAAgACAAIAAgACAAcABhAGQAZABpAG4AZwBUAG8AcAA6ACAAcwBwAGEAYwBlACwACgAgACAAIAAgACAAIAAgACAAcABhAGQAZABpAG4AZwBCAG8AdAB0AG8AbQA6ACAAcwBwAGEAYwBlACwACgAgACAAIAAgACAAIAAgACAAcABhAGQAZABpAG4AZwBMAGUAZgB0ADoAIABzAHAAYQBjAGUALAAKACAAIAAgACAAIAAgACAAIABwAGEAZABkAGkAbgBnAFIAaQBnAGgAdAA6ACAAcwBwAGEAYwBlACwACgAgACAAIAAgACAAIAAgACAAbQBhAHIAZwBpAG4AVABvAHAAOgAgAHMAcABhAGMAZQAsAAoAIAAgACAAIAAgACAAIAAgAG0AYQByAGcAaQBuAEIAbwB0AHQAbwBtADoAIABzAHAAYQBjAGUALAAKACAAIAAgACAAIAAgACAAIAB0AGUAeAB0AEEAbABpAGcAbgA6ACAAWwAnAGwAZQBmAHQAJwAsACAAJwBjAGUAbgB0AGUAcgAnACwAIAAnAHIAaQBnAGgAdAAnAF0ALAAKACAAIAAgACAAfQAsAAoAIAAgACAAIABzAGgAbwByAHQAaABhAG4AZABzADoAIAB7AAoAIAAgACAAIAAgACAAIAAgAHAAYQBkAGQAaQBuAGcAOgAgAFsAJwBwAGEAZABkAGkAbgBnAFQAbwBwACcALAAgACcAcABhAGQAZABpAG4AZwBCAG8AdAB0AG8AbQAnACwAIAAnAHAAYQBkAGQAaQBuAGcATABlAGYAdAAnACwAIAAnAHAAYQBkAGQAaQBuAGcAUgBpAGcAaAB0ACcAXQAsAAoAIAAgACAAIAAgACAAIAAgAHAAYQBkAGQAaQBuAGcAWAA6ACAAWwAnAHAAYQBkAGQAaQBuAGcATABlAGYAdAAnACwAIAAnAHAAYQBkAGQAaQBuAGcAUgBpAGcAaAB0ACcAXQAsAAoAIAAgACAAIAAgACAAIAAgAHAAYQBkAGQAaQBuAGcAWQA6ACAAWwAnAHAAYQBkAGQAaQBuAGcAVABvAHAAJwAsACAAJwBwAGEAZABkAGkAbgBnAEIAbwB0AHQAbwBtACcAXQAsAAoAIAAgACAAIAB9ACwACgB9ACkAOwA%3D

### Expected result

It shouldn't detect these as secrets.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by Exudev** [comment]
Hi! I'd like to work on this issue if it's still available.
Looking at the false positives, it seems the noSecrets rule is triggering on camelCase CSS property names like paddingBottom, paddingRight, and backgroundColor.
Possible approach:
I'm thinking we could improve the rule by adding context awareness for CSS property names. 

Could you assign this issue to me if the approach sounds good? @lloydjatkinson 

Thanks!

**Comment by dyc3** [comment]
I don't think that's a good solution. It treats the symptom not the problem, which is that the entropy calculation we use is not very good.

**Comment by Exudev** [comment]
> I don't think that's a good solution. It treats the symptom not the problem, which is that the entropy calculation we use is not very good.

I'd love to tackle improving the entropy calculation instead. Could you point me in the right direction?

**Comment by dyc3** [comment]
Unfortunately, I can only really give you pointers about the codebase. I'm not up to date on the state of the art for this problem. You'll have to research it yourself. IIRC, somebody else posted something about it in another issue I think. I'll see if I can find it for you.

**Comment by dyc3** [comment]
I can't find the comment. Github's search is a bit difficult sometimes.

**Comment by Exudev** [comment]
> I can't find the comment. Github's search is a bit difficult sometimes.

No worries, thanks for taking the time to look! I appreciate the guidance.
I'll do some research on entropy-based secret detection and see what improvements have been made in this space.

**Comment by lloydjatkinson** [comment]
You'll need to tag someone else about it, I'm just reporting the issue.

**Comment by Netail** [comment]
https://github.com/biomejs/biome/issues/7985 & https://github.com/biomejs/biome/issues/8136

Might be somewhat related
